### PR TITLE
Update snmpd_no_default_password

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- add tests

- make oval stricter

- add two xccdf variables for public and private community strings

- update Bash reemdiation

- add Ansible remediation

#### Rationale:

- stig effort

reference: https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-72313?version=v2r7

STIG is more stricter, it searches for all occurences of "private" and "public". Therefore I hardened the check.

I added two interactive variables which specify new public / private community strings.

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule snmpd_not_default_password:
 Bash remediation was deleted. No test for it will be selected.
 Text changed in OVAL check.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml snmpd_not_default_password

**Comment by redhatrises** [comment]
> The bash remediation searches for uncommented lines with "public" or Private" and comments them. I see this as an unexpected alteration of user configuration. I suggest to remove the remediation and let user fix it.

Let's update the remediation and not remove it.

**Comment by redhatrises** [comment]
> > > The bash remediation searches for uncommented lines with "public" or Private" and comments them. I see this as an unexpected alteration of user configuration. I suggest to remove the remediation and let user fix it.
> > 
> > 
> > Let's update the remediation and not remove it.
> 
> Well, I am not against, but how do you want to do that? We can't know what identifiers the user choses to replace "public" and "private"? Should we go with an XCCDF variable? If yes, can there be a variable without default value?

Yes, please use on XCCDF value. The default value could be something like "CHANGEME". We would need to also make sure that the value is interactive.

**Comment by mildas** [comment]
**Changes identified:**
Rule snmpd_not_default_password:
 Ansible remediation newly added.
 Found change in bash remediation.
 Text changed in OVAL check.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml snmpd_not_default_password
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml snmpd_not_default_password

**Comment by vojtapolasek** [comment]
I added interactive variables. Any tips where to specify that it is a very good idea to change this variables to something user-specific?

File: linux_os/guide/services/snmp/snmp_configure_server/snmpd_not_default_password/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  replace:
+    path: "/etc/snmp/snmpd.conf"
+    #regexp: '^[#](.*)public(.*)$'
+    regexp: 'public'
+    replace: '{{ var_snmpd_ro_string }}'
+
+- name: "Replace all instances of SNMP RW strings"
+  replace:
+    path: "/etc/snmp/snmpd.conf"
+    #regexp: '^[#](.*)private(.*)$'
```
Comment (redhatrises) [inline linux_os/guide/services/snmp/snmp_configure_server/snmpd_not_default_password/ansible/shared.yml:19]: @vojtapolasek do you want to remove this commented out line?

File: linux_os/guide/services/snmp/snmp_configure_server/snmpd_not_default_password/ansible/shared.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# strategy = configure
+# complexity = low
+# disruption = medium
+
+{{{ ansible_instantiate_variables("var_snmpd_ro_string", "var_snmpd_rw_string") }}}
+
+- name: "Replace all instances of SNMP RO strings"
+  replace:
+    path: "/etc/snmp/snmpd.conf"
+    #regexp: '^[#](.*)public(.*)$'
```
Comment (redhatrises) [inline linux_os/guide/services/snmp/snmp_configure_server/snmpd_not_default_password/ansible/shared.yml:12]: @vojtapolasek do you want to remove this commented out line?

**Comment by mildas** [comment]
**Changes identified:**
Rule snmpd_not_default_password:
 Found change in bash remediation.
 Ansible remediation newly added.
 Text changed in OVAL check.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml snmpd_not_default_password
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml snmpd_not_default_password

**Comment by vojtapolasek** [comment]
Oh thanks for noticing these forgotten lines, I removed them.

**Pull request merged**